app/src/ble.c

```diff
@@ -615,7 +615,21 @@ static int zmk_ble_init(const struct device *_arg) {
             LOG_ERR("Failed to delete setting: %d", err);
         }
     }
-#endif
+
+#if IS_ENABLED(CONFIG_ZMK_SPLIT_ROLE_CENTRAL)
```

Comment:
The preceding `ble/profiles/%d` should also only exist on central if I'm not mistaken.

For both, I'd also consider looping to some fixed integer (such as 16) and removing this `#if IS_ENABLED` entirely. This way the same reset file can be used for both halves and there's no orphaned settings if user decides to swap the controllers for central and peripheral.

Reply:
Oh, the peripherals only use `peripheral.c` which already has different bond clearing on the start, so really the checks are *right* since `ble.c` is only run on non-splits or split centrals.

Reply:
I'm not sure it's worth disrupting that much just to clear unused/dangling settings in the swap scenario.

Reply:
Ah yes, forgot about the fact that there are non-split keyboards.

Reply:
Actually, moved to your suggestion, since the settings_reset shield isn't built as a split at all.